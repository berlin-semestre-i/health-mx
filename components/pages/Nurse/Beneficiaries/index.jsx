import React, { PureComponent } from 'react'
import { Grid, Form, Button, Table } from 'semantic-ui-react'
import { DateInput } from 'semantic-ui-calendar-react'
import Header from '../../../shared/PageHeader'
import styled from 'styled-components'
import media from 'styled-media-query'
import Link from 'next/link'

const specialtiesOptions = [{
  text: 'Especialidad 1', value: 0,
}, {
  text: 'Especialidad 2', value: 1,
}]
const searchResults = [{
  beneficiary: 'Orduño Ahumada, Julia Paola',
  nextAppointment: '10 oct, 2018 - 8:00 a.m.',
  specialty: 'Medicina familiar',
  status: 'Regular',
},{
  beneficiary: 'Orduño Ahumada, Julia Paola',
  nextAppointment: '10 oct, 2018 - 8:00 a.m.',
  specialty: 'Medicina familiar',
  status: 'Regular',
},{
  beneficiary: 'Orduño Ahumada, Julia Paola',
  nextAppointment: '10 oct, 2018 - 8:00 a.m.',
  specialty: 'Medicina familiar',
  status: 'Regular',
}]

class Beneficiaries extends PureComponent {

  state = {
    date: '',
    beneficiaryName:'',
    specialty: '',
  }

  handleChange = (e, {name, value}) => {
    this.setState({ [name]: value })
  }

  render() {
    const { date, beneficiaryName, specialty } = this.state

    return (
      <React.Fragment>
        <Grid>
          <Header subtitle="Toma de somatometrías previas" title="Derechohabientes" />
          <Grid.Row>
            <Grid.Column>
              <Form>
                <h4>Búsqueda de derechohabientes</h4>
                <NurseFormGroup>
                  <Form.Input
                    width={7}
                    placeholder="Nombre"
                    value={beneficiaryName}
                    onChange={this.handleChange}
                    name="beneficiaryName"
                  />
                  <Form.Field width={4}>
                    <DateInput
                      placeholder="Fecha de cita"
                      iconPosition="left"
                      value={date}
                      name="date"
                      onChange={this.handleChange}
                    />
                  </Form.Field>
                  <Form.Dropdown
                    selection
                    fluid
                    width={3}
                    options={specialtiesOptions}
                    placeholder="Especialidad"
                    value={specialty}
                    name="specialty"
                    onChange={this.handleChange}
                  />
                  <Form.Field width={2}>
                    <Button content="Buscar" />
                  </Form.Field>
                </NurseFormGroup>
              </Form>
            </Grid.Column>
          </Grid.Row>
          <Grid.Row>
            <Grid.Column>
              <Table striped>
                <TableHeader>
                  <Table.Row>
                    <Table.HeaderCell>Derechohabiente</Table.HeaderCell>
                    <Table.HeaderCell>Próxima Cita</Table.HeaderCell>
                    <Table.HeaderCell>Área de especialidad</Table.HeaderCell>
                    <Table.HeaderCell>Estado</Table.HeaderCell>
                    <Table.HeaderCell className="empty"></Table.HeaderCell>
                  </Table.Row>
                </TableHeader>
                <Table.Body>
                  {searchResults.map((result, index) => (
                    <Table.Row key={index}>
                      <TableCell>{result.beneficiary}</TableCell>
                      <TableCell>{result.nextAppointment}</TableCell>
                      <TableCell>{result.specialty}</TableCell>
                      <TableCell>{result.status}</TableCell>
                      <TableCell>
                        <Link href="/nurse/consultation">
                          <b><a href="">Ver detalles</a></b>
                        </Link>
                      </TableCell>
                    </Table.Row>
                  ))}
                </Table.Body>
              </Table>
            </Grid.Column>
          </Grid.Row>
        </Grid>
      </React.Fragment>
    )
  }
}

export default Beneficiaries

const NurseFormGroup = styled(Form.Group)`
  & div.field .ui.button {
    width: 100%;
  }

  ${media.lessThan('small')`
    &&&.fields>div.field {
      margin-bottom: 1rem;
    }
  `}

  ${media.between('small', 'medium')`
    &.fields .seven.wide.field {
      width: 37% !important;
    }

    &.fields .four.wide.field {
      width: 20% !important;
    }

    &.fields .three.wide.field {
      width: 23.75% !important;
    }

    &.fields .two.wide.field {
      width: 18.5% !important;
    }
  `}
`
const TableCell = styled(Table.Cell)`
  &&& {
    padding: 1.1rem 1.5rem;
    border: none;
  }
`
const TableHeader = styled(Table.Header)`
 &&&& th {
   background-color: #808AA2;
   color: #FFFF;
   text-transform: uppercase;
   font-size: 12px;
   font-weight: 500;
   padding: 1.1rem 1.5rem;
   border: none;
 }

 ${media.lessThan('small')`
    &&&& tr {
      padding: 0;
    }

    &&&& tr>th.empty {
      display: none !important;
    }
  `}
`