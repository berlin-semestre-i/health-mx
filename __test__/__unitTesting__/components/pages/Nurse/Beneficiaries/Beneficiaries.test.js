import React from 'react'
import { shallow, mount } from 'enzyme'
import Beneficiaries from '../../../../../../components/pages/Nurse/Beneficiaries/index'

describe('Beneficiaries', () => {
  it('renders', () => {
    const component = shallow(<Beneficiaries />)
    expect(component).toMatchSnapshot()
  })

  it('updates state on input change', () => {
    const wrapper = mount(<Beneficiaries />)
    wrapper.find('input').forEach(node => {
      let name = node.prop('name')
      node.simulate('change', {target: {value: 'New Value'}})
      expect(wrapper.state(name)).toBe('New Value')
    })
  })
})