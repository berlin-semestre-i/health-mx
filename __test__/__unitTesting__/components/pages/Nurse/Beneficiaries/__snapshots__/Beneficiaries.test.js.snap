// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Beneficiaries renders 1`] = `
<Fragment>
  <Grid>
    <PageHeader
      subtitle="Toma de somatometrías previas"
      title="Derechohabientes"
    />
    <GridRow>
      <GridColumn>
        <Form
          as="form"
        >
          <Beneficiaries__NurseFormGroup>
            <FormInput
              as={[Function]}
              control={[Function]}
              label="Búsqueda de derechohabientes"
              name="beneficiaryName"
              onChange={[Function]}
              placeholder="Nombre"
              value=""
              width={7}
            />
            <FormField
              className="no-label"
              width={4}
            >
              <DateInput
                dateFormat="DD-MM-YYYY"
                iconPosition="left"
                name="date"
                onChange={[Function]}
                placeholder="Fecha de cita"
                preserveViewMode={true}
                startMode="day"
                value=""
              />
            </FormField>
            <FormDropdown
              as={[Function]}
              className="no-label"
              control={[Function]}
              fluid={true}
              name="specialty"
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "text": "Especialidad 1",
                    "value": 0,
                  },
                  Object {
                    "text": "Especialidad 2",
                    "value": 1,
                  },
                ]
              }
              placeholder="Especialidad"
              selection={true}
              value=""
              width={3}
            />
            <FormField
              className="no-label"
              width={2}
            >
              <Button
                as="button"
                content="Buscar"
                role="button"
              />
            </FormField>
          </Beneficiaries__NurseFormGroup>
        </Form>
      </GridColumn>
    </GridRow>
    <GridRow>
      <GridColumn>
        <Table
          as="table"
          striped={true}
        >
          <Beneficiaries__TableHeader>
            <TableRow
              as="tr"
              cellAs="td"
            >
              <TableHeaderCell
                as="th"
              >
                Derechohabiente
              </TableHeaderCell>
              <TableHeaderCell
                as="th"
              >
                Próxima Cita
              </TableHeaderCell>
              <TableHeaderCell
                as="th"
              >
                Área de especialidad
              </TableHeaderCell>
              <TableHeaderCell
                as="th"
              >
                Estado
              </TableHeaderCell>
              <TableHeaderCell
                as="th"
                className="empty"
              />
            </TableRow>
          </Beneficiaries__TableHeader>
          <TableBody
            as="tbody"
          >
            <TableRow
              as="tr"
              cellAs="td"
              key="0"
            >
              <Beneficiaries__TableCell>
                Orduño Ahumada, Julia Paola
              </Beneficiaries__TableCell>
              <Beneficiaries__TableCell>
                10 oct, 2018 - 8:00 a.m.
              </Beneficiaries__TableCell>
              <Beneficiaries__TableCell>
                Medicina familiar
              </Beneficiaries__TableCell>
              <Beneficiaries__TableCell>
                Regular
              </Beneficiaries__TableCell>
              <Beneficiaries__TableCell>
                <Link
                  href="/nurse/consultation"
                >
                  <b>
                    <a
                      href=""
                    >
                      Ver detalles
                    </a>
                  </b>
                </Link>
              </Beneficiaries__TableCell>
            </TableRow>
            <TableRow
              as="tr"
              cellAs="td"
              key="1"
            >
              <Beneficiaries__TableCell>
                Orduño Ahumada, Julia Paola
              </Beneficiaries__TableCell>
              <Beneficiaries__TableCell>
                10 oct, 2018 - 8:00 a.m.
              </Beneficiaries__TableCell>
              <Beneficiaries__TableCell>
                Medicina familiar
              </Beneficiaries__TableCell>
              <Beneficiaries__TableCell>
                Regular
              </Beneficiaries__TableCell>
              <Beneficiaries__TableCell>
                <Link
                  href="/nurse/consultation"
                >
                  <b>
                    <a
                      href=""
                    >
                      Ver detalles
                    </a>
                  </b>
                </Link>
              </Beneficiaries__TableCell>
            </TableRow>
            <TableRow
              as="tr"
              cellAs="td"
              key="2"
            >
              <Beneficiaries__TableCell>
                Orduño Ahumada, Julia Paola
              </Beneficiaries__TableCell>
              <Beneficiaries__TableCell>
                10 oct, 2018 - 8:00 a.m.
              </Beneficiaries__TableCell>
              <Beneficiaries__TableCell>
                Medicina familiar
              </Beneficiaries__TableCell>
              <Beneficiaries__TableCell>
                Regular
              </Beneficiaries__TableCell>
              <Beneficiaries__TableCell>
                <Link
                  href="/nurse/consultation"
                >
                  <b>
                    <a
                      href=""
                    >
                      Ver detalles
                    </a>
                  </b>
                </Link>
              </Beneficiaries__TableCell>
            </TableRow>
          </TableBody>
        </Table>
      </GridColumn>
    </GridRow>
  </Grid>
</Fragment>
`;
